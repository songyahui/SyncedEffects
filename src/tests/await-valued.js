"use hopscript"

var hh = require("hiphop");

function foo(evt) {
   console.log("foo called by", evt.type, "with value", evt.nowval);
}

var inSig = {accessibility: hh.IN};
var outSig = {accessibility: hh.OUT};

var prg = <hh.module I=${inSig} O=${outSig}>
    <hh.await I />
    <hh.emit O apply=${function() {return this.I.nowval}}/>
</hh.module>;

var m = new hh.ReactiveMachine(prg, "awaitvalued");
m.addEventListener("O", foo);

exports.prg = m;
