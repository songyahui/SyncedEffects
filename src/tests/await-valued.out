awaitvalued> ;
--- Output:
awaitvalued> I(23);
foo called by O with value 23
--- Output: O(23)
awaitvalued> ;
--- Output:
